### PR TITLE
Update v5 dependencies to latest versions - Part 1

### LINKED ISSUE #6174: Update 3rd party dependencies to their latest versions in v5
## Summary

As part of the v5 release, we will update the 3rd party dependencies to their latest supported version. 

To simplify maintenance of 3rd party versions, we should move the required version to the props file. 

TODO: Gabo to spec sample for this

### PR DESCRIPTION
**Pull request by BruceHaley**
Fixes #6174

## Description
Update all 3rd party dependencies whose latest versions require no code fixups.
Normalize dependency reference to Directory.Build.props file.
Add 'dir workspace' diagnostics to .yml pipelines.

Parts 2 and later will update dependencies that require code fixups.

Dependencies will not be updated to 6.0. Those updates will be left for issue #5977.

### CODE REVIEW

File: libraries/Microsoft.Bot.Builder.AI.LUIS/Microsoft.Bot.Builder.AI.Luis.csproj [hunk 1]
```diff
@@ -30,9 +30,9 @@
   </ItemGroup>
 
   <ItemGroup>
-    <PackageReference Include="Microsoft.Extensions.Configuration" Version="2.1.0" />
-    <PackageReference Include="Microsoft.Extensions.Configuration.Json" Version="2.1.0" />
-    <PackageReference Include="Newtonsoft.Json" Version="12.0.3" />
+    <PackageReference Include="Microsoft.Extensions.Configuration" Version="6.0.0" />
```
Comment (gabog) [inline libraries/Microsoft.Bot.Builder.AI.LUIS/Microsoft.Bot.Builder.AI.Luis.csproj:33]: I "think" we should target 3.1 here, not sure if 6.0 will work for aspnet core projects targeting 3.1 (similar situation with all the other places where we updated 2.1 to 6.0)

File: testbots/Directory.Build.props [hunk 2]
```diff
@@ -10,6 +10,10 @@
     <NoWarn>$(NoWarn);SA0001;CS1573;CS1591;CS1712;SX1309</NoWarn>
   </PropertyGroup>
 
+  <ItemGroup Condition="'$(TargetFramework)' != 'netstandard2.0'">
```
Comment (gabog) [inline testbots/Directory.Build.props:13]: Let's keep explicit references to newtonsoft on each project, Muthu is working on trying to see if we can remove this dependency for v5

**Review by gabog — COMMENTED** [review]
See comments and review them with Muthu

File: libraries/Microsoft.Bot.Builder/Microsoft - Backup.Bot.Builder.csproj [hunk 3]
```diff
@@ -0,0 +1,38 @@
+﻿<Project Sdk="Microsoft.NET.Sdk">
```
Comment (msomanathan) [inline libraries/Microsoft.Bot.Builder/Microsoft - Backup.Bot.Builder.csproj:1]: Why is this csproj needed?

File: libraries/Microsoft.Bot.Builder.AI.Orchestrator/Microsoft.Bot.Builder.AI.Orchestrator.csproj [hunk 4]
```diff
@@ -35,5 +35,4 @@
   <ItemGroup>
     <ProjectReference Include="..\Microsoft.Bot.Builder.Dialogs\Microsoft.Bot.Builder.Dialogs.csproj" />
   </ItemGroup>
-
```
Comment (msomanathan) [inline libraries/Microsoft.Bot.Builder.AI.Orchestrator/Microsoft.Bot.Builder.AI.Orchestrator.csproj:38]: No changes in this file

Comment (BruceHaley) [reply libraries/Microsoft.Bot.Builder.AI.Orchestrator/Microsoft.Bot.Builder.AI.Orchestrator.csproj:38]: Formatting fix.

File: FunctionalTests/Microsoft.Bot.Builder.FunctionalTests/Microsoft.Bot.Builder.FunctionalTests.csproj [hunk 5]
```diff
@@ -3,6 +3,9 @@
   <PropertyGroup>
     <TargetFramework Condition="'$(BuildTarget)' == 'netcoreapp31'">netcoreapp3.1</TargetFramework>
     <TargetFrameworks Condition="'$(BuildTarget)' == ''">netcoreapp3.1</TargetFrameworks>
+    <!-- Next line suppresses error NETSDK1073: The FrameworkReference 'Microsoft.WindowsDesktop.App.WindowsForms' was not recognized -->
```
[the remark this answers is not in the document]
Comment (BruceHaley) [reply FunctionalTests/Microsoft.Bot.Builder.FunctionalTests/Microsoft.Bot.Builder.FunctionalTests.csproj:6]: This error suppression was the suggested fix in StackOverflow. This error occurs on the Mac using dotnet build, does not occur in Windows using MSBuild. I will dig in to discover the cause.

**Review by msomanathan — APPROVED** [review]

**Pull request merged**